shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -685,8 +638,11 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
   auto did_submit = true;
   auto num_platform_views = composition_order_.size();
 
-  // TODO(hellohuanlin) this double for-loop is expensive with wasted computations.
-  // See: https://github.com/flutter/flutter/issues/145802
+  size_t missing_layer_count = 0;
```

Comment:
With this design, if we're missing an overlay layer then it won't render until the next frame. I think we can handle that by forcing resubmission of the frame but I'll have to follow up on it.